src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,85 @@
+package duke.main;
+
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+import duke.error.DukeException;
+import duke.error.InsufficientArgumentError;
+import duke.error.MissingKeywordError;
+import duke.error.UnknownCommandError;
+
+import java.util.Date;
+
+public class Parser {
+    public static Command parse(String input) throws DukeException {
+        String[] currentInputArray = input.split("\\s");
+        switch (currentInputArray[0]) {
+        case "list":
+            return new ListCommand();
+        case "done":
+            if (currentInputArray.length < 2) {
```

Comment:
You may want to consider shifting the if cases to a separate method that handles checking of valid commands - reduce repeated codes in the switch block for eg. line 35